appserver/web/gf-weld-connector/src/main/java/org/glassfish/weld/connector/WeldUtils.java

```diff
@@ -403,20 +368,29 @@ private static boolean isCDIEnablingAnnotation(AnnotationType annotationType,
     public static boolean hasValidAnnotation(Class              annotatedClass,
```

Comment:
Could this method signature be changed to accept `Set<String>` painlessly? This would make copying the collections obsolete.

Reply:
There are placese where constant fields are passed, so we should not modify them. The copy is done on public api level and collections passed as parameter are only local implementation which is safe to verify their modification.

Reply:
Ok, I see.